dali/image/image_factory.cc

```diff
@@ -17,7 +17,11 @@
 #include "dali/image/png.h"
 #include "dali/image/bmp.h"
 #include "dali/image/jpeg.h"
+#ifdef DALI_USE_LIBTIFF
+#include "dali/image/tiff_libtiff_impl.h"
+#else
```

Comment:
I don't think this `ifdef` should be neccessary. I mean, the header file could (I guess) compile regardless of presence of libtiff, while any compile-flag specific stuff can be hidden in `tiff_libitff.cc`

Reply:
I disagree. If we are not going to use the impl, I'd not include anything